File: gym_fe/src/type/schedule.ts
export interface Schedule {
  scheduleID: string;
  dayOfWeek: string;
  startTime: string;
  endTime: string;
  maxParticipants: number;
}

export interface ScheduleFormData {
  dayOfWeek: string;
  maxParticipants: number;
  startTime: string;
  endTime: string;
}

export interface ScheduleApiResponse {
  items: Schedule[];
  totalPages: number;
  currentPage: number;
  totalItems: number;
}

File: gym_be/gym_be/Models/Entities/Schedule.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace gym_be.Models.Entities
{
    [Table("schedule")]
    public class Schedule
    {
        [Key]
        [Column("scheduleid")]
        public Guid ScheduleID { get; set; }

        [Column("dayofweek")]
        public string DayOfWeek { get; set; }

        [Column("maxparticipants")]
        public int MaxParticipants { get; set; }

        [Column("starttime")]
        [MaxLength(5)]
        public string StartTime { get; set; }  

        [Column("endtime")]
        [MaxLength(5)]
        public string EndTime { get; set; }    

        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}
